ast-visual-studio-extension/CxExtension/Services/ASCAService.cs

```diff
@@ -0,0 +1,490 @@
+                        selection.MoveToLineAndOffset(detail.Line, 1);
+                        selection.SelectLine();
+                    };
+
+                    _errorListProvider.Tasks.Add(task);
+
+                    // Create visual marker
+                    try
+                    {
+
+                        // Find the problematic text position
+                        string problemTextValue = detail.ProblematicLine;
+                        int startIndex = problemTextValue.Length - problemTextValue.TrimStart().Length;
+                        if (startIndex == -1)
+                        {
+                            startIndex = 0;
+                        }
+                        int endIndex = startIndex + problemTextValue.Length;
+
+                        IVsTextLineMarker[] markers = new IVsTextLineMarker[1];
+                        var errorSpan = new TextSpan
+                        {
+                            iStartLine = detail.Line - 1,
+                            iStartIndex = startIndex,
+                            iEndLine = detail.Line - 1,
+                            iEndIndex = endIndex
+                        };
+
+                        var markerClient = new VsTextMarkerClient(detail.RuleName, detail.RemediationAdvise, detail.Severity);
+                        hr = buffer.CreateLineMarker(
+                            (int)GetMarkerType(detail.Severity),
+                            errorSpan.iStartLine,
+                            errorSpan.iStartIndex,
+                            errorSpan.iEndLine,
+                            errorSpan.iEndIndex,
+                            markerClient,
+                            markers
+                        );
+
+                        if (ErrorHandler.Succeeded(hr) && markers[0] != null)
+                        {
+                            _activeMarkers.Add(markers[0]);
+                        }
+                        else
+                        {
+                            Debug.WriteLine($"Failed to create marker on line {detail.Line}");
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Debug.WriteLine($"Failed to create marker on line {detail.Line}: {ex.Message}");
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine($"Failed to setup text view: {ex.Message}");
+            }
+
+        }
+
+        private class VsTextMarkerClient : IVsTextMarkerClient
+        {
+            private readonly string _ruleName;
+            private readonly string _remediation;
+
+            public VsTextMarkerClient(string ruleName, string remediation, string severity)
+            {
+                _ruleName = ruleName;
+                _remediation = remediation;
+            }
+
+            public void MarkerInvalidated()
```

Comment:
No need for virtual/override modifiers because:
* no need for `virtual` as this is an interface implementation, not meant for inheritance
* no need for `override` as it's not overriding an existing method